iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/EnvSecretCheck.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.checks;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.sonar.check.Rule;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.docker.tree.api.EnvTree;
+import org.sonar.iac.docker.tree.api.KeyValuePairTree;
+
+@Rule(key = "S6472")
+public class EnvSecretCheck implements IacCheck {
+
+  private static final String MESSAGE = "Make sure that using ENV to handle a secret is safe here.";
+
+  private static final List<String> ENTITIES = List.of("ACCESS", "AMPLITUDE", "ANSIBLE", "ADMIN", "API",
+    "APP", "AUTH", "CLIENT", "CONFIG", "DATABASE", "DB", "ENCRYPTION", "ENV", "FACEBOOK", "FIREBASE", "FTP", "GIT",
+    "GITHUB", "GITLAB", "HONEYCOMB", "JWT", "KEYCLOAK", "KEYRING", "LDAP", "MAIL", "MASTER", "MARIADB", "MSSQL",
+    "MYSQL", "NPM", "OAUTH", "OAUTH2", "PG", "POSTGRES", "REDIS", "REFRESH", "REPLICATION", "ROOT", "RPC", "SA",
+    "SECRET", "SERVER", "SIGN", "SIGNING", "SLACK", "SVN", "USER", "VNC", "WEBHOOK");
+
+  private static final List<String> SECRETS = List.of("CREDENTIALS", "KEY", "PASS", "PASSPHRASE", "PASSWD", "PASSWORD",
+    "SECRET", "TOKEN");
+
+  private static final List<String> EXCLUSIONS = List.of("ALLOW", "DIR", "EXPIRE", "EXPIRY", "FILE", "ID",
+    "LOCATION", "NAME", "OWNER", "PATH", "URL");
+
+  private static final Pattern UPPERCASE_UNDERSCORE_NAME_PATTERN = Pattern.compile("^[A-Z_]+$");
+  private static final Pattern UPPERCASE_DASH_NAME_PATTERN = Pattern.compile("^[A-Z-]+$");
+
+  private static final Pattern CAMELCASE_NAME_PATTERN = Pattern.compile("^[A-Za-z]+$");
+
+  private static final Pattern CAMELCASE_SPLIT_PATTERN = Pattern.compile("(?<!(^|[A-Z]))(?=[A-Z])|(?<!^)(?=[A-Z][a-z])");
```

Comment:
🤯 Maybe the reference to https://stackoverflow.com/questions/7593969/regex-to-split-camelcase-or-titlecase-advanced or some comment is needed here?